api/src/main/java/io/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -44,13 +47,26 @@ public FileIteratingFirehose(
   {
     this.lineIterators = lineIterators;
     this.parser = parser;
+    final ParseSpec parseSpec = parser.getParseSpec();
+    if (parseSpec instanceof CSVParseSpec) {
```

Comment:
Ah, I see.. But I'm still not sure that's a better way. I prefer to not use nulls for representing something meaningful because it is difficult to check that a null value is an expected result or an error. Is it for sharing some common code paths?